File: Queue/Task_scheduling.py
class Solution:
    # @param A : list of integers
    # @param B : list of integers
    # @return an integer
    def solve(self, A, B):
        cnt = 0
        for i in B:
            x = A.index(i)
            if x == 0:
                cnt += 1
                A.pop(0)
            else:
                A = A[x:]+A[:x]
                cnt += x+1
                A.pop(0)            
        return cnt

File: Queue/Task_scheduling.java
public class Solution {
    public int solve(ArrayList<Integer> A, ArrayList<Integer> B) {
        int n=A.size();
        Queue<Integer>q=new LinkedList<Integer>();
        int i;
        for(i=0;i<n;i++){
            q.add(A.get(i));
        }
        int j=0;
        int cycles=0;
        for(j=0;j<n;j++){
            while(B.get(j)!=q.peek()){
                int d=q.remove();
                q.add(d);
                cycles+=1;
            }
            q.remove();
            cycles+=1;
            
        }
        return cycles;

    }
}
